community/kernel/src/main/java/org/neo4j/kernel/recovery/Recovery.java

```diff
@@ -82,11 +99,27 @@ public void init() throws Throwable
         }
 
         monitor.recoveryRequired( recoveryFromPosition );
+        spi.startRecovery();
+
+        // Backwards for neo store only
+        long checkpointTxId = TransactionIdStore.BASE_TX_ID;
+        try (   TransactionCursor transactionsToRecover = spi.getTransactionsInReverseOrder( recoveryFromPosition );
+                RecoveryApplier recoveryVisitor = spi.getRecoveryApplier( REVERSE_RECOVERY ); )
+        {
+            while ( transactionsToRecover.next() )
+            {
+                recoveryVisitor.visit( transactionsToRecover.get() );
+                checkpointTxId = transactionsToRecover.get().getCommitEntry().getTxId();
+            }
+        }
+
+        monitor.reverseStoreRecoveryCompleted( checkpointTxId );
 
+        // Forward with all appliers
         LogPosition recoveryToPosition;
         CommittedTransactionRepresentation lastTransaction = null;
-        Visitor<CommittedTransactionRepresentation,Exception> recoveryVisitor = spi.startRecovery();
-        try ( TransactionCursor transactionsToRecover = spi.getTransactions( recoveryFromPosition ) )
+        try (   TransactionCursor transactionsToRecover = spi.getTransactions( recoveryFromPosition );
+                RecoveryApplier recoveryVisitor = spi.getRecoveryApplier( RECOVERY ); )
```

Comment:
Alright, now we got no more spaces to spare, used up the entire buffer of spaces we had. And winter is coming... How will we manage.

Reply:
No likey my feng shui?

Reply:
Copy from above: Auto format will remove the extra spaces from both lines. At least for me 😄  doesn't work for you?

Reply:
I'd guess we're passed point of diminishing returns when discussing spaces, but will change the spaces for this one

Reply:
`Ctrl+Shift+F` in Eclipse puts them aligned and in the right place it seems :)